numpy/lib/_function_base_impl.py

```diff
@@ -5277,8 +5277,8 @@ def meshgrid(*xi, copy=True, sparse=False, indexing='xy'):
 
     if indexing == 'xy' and ndim > 1:
         # switch first and second axis
-        output[0].shape = (1, -1) + s0[2:]
-        output[1].shape = (-1, 1) + s0[2:]
+        output[0] = output[0].reshape((1, -1) + s0[2:])
```

Comment:
I'd tend to instead generate `output` separately for this case, but obviously no big deal.